librz/bin/filter.c

```diff
@@ -80,24 +80,25 @@ RZ_API char *rz_bin_filter_name(RzBinFile *bf, HtPU *db, ut64 vaddr, char *name)
 
 RZ_API void rz_bin_filter_sym(RzBinFile *bf, HtPP *ht, ut64 vaddr, RzBinSymbol *sym) {
 	rz_return_if_fail(ht && sym && sym->name);
-	const char *name = sym->name;
-	// if (!strncmp (sym->name, "imp.", 4)) {
-	// demangle symbol name depending on the language specs if any
-	if (bf && bf->o && bf->o->lang) {
-		const char *lang = rz_bin_lang_tostring(bf->o->lang);
-		char *dn = rz_bin_demangle(bf, lang, sym->name, sym->vaddr, false);
-		if (dn && *dn) {
+	const char *name = sym->dname ? sym->dname : sym->name;
+
+	if (bf && bf->o && bf->o->lang && !sym->dname) {
+		char *dn = rz_bin_demangle(bf, NULL, name, sym->vaddr, false);
+		if (RZ_STR_ISNOTEMPTY(dn)) {
 			sym->dname = dn;
-			// XXX this is wrong but is required for this test to pass
-			// pmb:new pancake$ bin/rz_test.js db/formats/mangling/swift
-			sym->name = dn;
 			// extract class information from demangled symbol name
 			char *p = strchr(dn, '.');
 			if (p) {
 				if (IS_UPPER(*dn)) {
+					if (sym->classname) {
+						free(sym->classname);
+					}
```

Comment:
```suggestion
					free(sym->classname);
```
same few lines below